azurelinuxagent/ga/policy/regorus.py

```diff
@@ -0,0 +1,146 @@
+    Any errors thrown in this class should be caught and handled by PolicyEngine.
+    """
+
+    def __init__(self, policy_file, rule_file):
+        """
+        Rule_file is expected to point to a valid Regorus file.
+
+        Policy_file should be a path to a valid JSON policy (data) file.
+        The expected file format is:
+        {
+            "azureGuestAgentPolicy": {
+                "policyVersion": "0.1.0",
+                "signingRules": {
+                    "extensionSigned": <true, false>
+                },
+                "allowListOnly": <true, false>
+            },
+            "azureGuestExtensionsPolicy": {
+                "allowed_ext_1": {
+                    "signingRules": {
+                        "extensionSigned": <true, false>
+                    }
+                }
+        }
+        """
+        self._engine = None
+        self._rule_file = rule_file
+        self._policy_file = policy_file
+
+
+    @property
+    def policy_file(self):
+        return self._policy_file
+    
+    @property
+    def rule_file(self):
+        return self._rule_file
+
+    def eval_query(self, input_to_check, query):
+        """
+        Input_to_check should be type dict.
+        Example format for extension policy:
+        {
+            "extensions": {
+                "<extension_name_1>": {
+                    "signingInfo": {
+                        "extensionSigned": <true, false>
+                    }
+                },
+                "<extension_name_2>": {
+                    "signingInfo": {
+                        "extensionSigned": <true, false>
+                    }
+                }, ...
+        }
+
+        In this method, we call the Regorus executable via run_command to query the policy engine.
+
+        Command:
+            regorus eval -d <rule_file.rego> -d <policy_file.json> -i <input_file.json> <QUERY>
+
+        Parameters:
+            -d, --data <rule.rego|policy.json> : Rego file or JSON file.
+            -i, --input <input.json> : Input file in JSON format.
+            <QUERY>  Query. Rego query block in the format "data.<optional_query>"
+
+        Return Codes:
+            0 - successful query. optional parameters may be missing
+            1 - file error: unsupported file type, error parsing file, file not found
+                ex: "Error: Unsupported data file <file>. Must be rego or json."
+                ex: "Error: Failed to read <file>. No such file or directory."
+            2 - usage error: missing query argument, unexpected or unlabeled parameter
+                ex: "Error: the following required arguments were not provided: <QUERY>"
+                ex: "Error: Unexpected argument <arg> found."
+        """
+        # Write input_to_check to a temp file, because Regorus requires input to be a file path.
+        # Tempfile is automatically cleaned up at the end of with block
```

Comment:
Is there a typo here? 

> Tempfile is automatically cleaned up at the end of (?) with block

Reply:
No typo, the tempfile is cleaned up at the end of the "with" block